### ISSUE TITLE
MPVPaper support?

### ISSUE THREAD

**Issue by Continous** [anufrievroman/waypaper#12]
MPV Paper (https://github.com/GhostNaN/mpvpaper) is a wallpaper client that utilizes MPV to play a video on the user's wallpaper. Apparently it is easy to control and forward files/inputs to it. Videos are a lot heavier on people's system, but it may be preferable or more convenient than a simple animated image format for some.

**Comment by anufrievroman** [comment]
One problem that I faced last time I considered it was that we need to generate previews of the videos (which is more difficult to efficiently) instead of previews of images (which is done with standard GTK tools). Plus, it does not seem to support all other features like scaling, background colors etc. But that is something I can try to implement.

**Comment by Continous** [comment]
> One problem that I faced last time I considered it was that we need to generate previews of the videos (which is more difficult to efficiently) instead of previews of images (which is done with standard GTK tools). Plus, it does not seem to support all other features like scaling, background colors etc. But that is something I can try to implement.

Everyone appreciates your effort, so don't burn yourself out on it. Just take your time. I don't think anyone would assume it to be easy.

**Comment by anufrievroman** [comment]
The support is implemented thanks to the efforts by @CalfMoon , it will be available in the next version (or try it in the current main branch).

**Comment by anufrievroman** [comment]
The one issue I see with current implementation of mpvpaper support is that mpvpaper is not a daemon, it just creates a new process every time you run it. So, when we change a wallpaper, old ones keep running. If we kill the old ones before setting the new wallpaper, then it only works if one wallpaper is set for all monitors. I mean, I don't see a way to track which process is for which monitor to kill it next time we launch a new process for this monitor.

**Comment by CalfMoon** [comment]
I sadly didn't consider that. Maybe there is a workaround. We should probably postpond mpvpaper for the next release i will look into it in a week or two.

**Comment by anufrievroman** [comment]
I'm thinking to still release it but without monitor selection at the moment, only for "All", with killing previous instance every time, unless I find some trick this weekend :)

**Comment by anufrievroman** [comment]
Okay, thanks to the help from mpvpaper developer, I figured out how to properly interact with mpvpaper.
Now, I made it in such a way to ensure that it works for all monitors (not specific ones). If anyone is intersted, the way it works is this: we launch mpvpaper in a specific "socket" custom named as `mpv-socket-<monitor>`, then we can change which video is playing in that specific socket. Actually, we can even implement other controls, like pause video etc.

The only remaining problem is how to check on which sockets (i.e. on which monitors) mpvpaper is already running, and then either to start a new mpvpaper process or just call an existing socket.
